stock_by_warehouse/tests/test_stock_available_unreserved.py

```diff
@@ -0,0 +1,124 @@
+        self.supplier_location = self.env.ref('stock.stock_location_suppliers')
+        self.stock_location = self.env.ref(
+            'stock.stock_location_stock')
+        self.customer_location = self.env.ref('stock.stock_location_customers')
+        self.uom_unit = self.env.ref('product.product_uom_unit')
+
+        # Create product template
+        self.template_ab = self.template_obj.create(
+            {'name': 'templAB',
+             'uom_id': self.uom_unit.id,
+             })
+
+        self.product_values = {'name': 'product A',
+                               'standard_price': 1,
+                               'type': 'product',
+                               'uom_id': self.uom_unit.id,
+                               'default_code': 'A',
+                               'product_tmpl_id': self.template_ab.id,
+                               }
+
+    def create_product(self, values):
+        return self.product_obj.create(values)
+
+    def create_picking(self, picking_type, loc_orig, loc_dest, product, qty):
+        picking = self.picking_obj.create({
+            'picking_type_id': picking_type,
+            'location_id': loc_orig.id,
+            'location_dest_id': loc_dest.id,
+            'move_lines': [
+                (0, 0, {
+                    'name': 'Test move',
+                    'product_id': product.id,
+                    'product_uom': product.uom_id.id,
+                    'product_uom_qty': qty,
+                    'location_id': loc_orig.id,
+                    'location_dest_id': loc_dest.id,
+                })]
+        })
+        return picking
+
+    def test01_stock_levels(self):
+        """checking that qty_available_not_res actually reflects \
+        the variations in stock, both on product and template"""
+
+        # Create product A and B
+
+        product_a = self.create_product(self.product_values)
+        product_b = product_a.copy({'name': 'product B',
+                                    'default_code': 'B',
+                                    'product_tmpl_id': self.template_ab.id})
+
+        # Create a picking move from INCOMING to STOCK
+
+        picking_in_a = self.create_picking(self.ref('stock.picking_type_in'),
+                                           self.supplier_location,
+                                           self.stock_location,
+                                           product_a, 2)
+
+        picking_in_b = self.create_picking(self.ref('stock.picking_type_in'),
+                                           self.supplier_location,
+                                           self.stock_location,
+                                           product_b, 3)
+
+        def compare_qty_available_not_res(product, value):
+            # Refresh, because the function field is not recalculated between
+            # transactions
+            product.refresh()
+            self.assertEqual(product.qty_available_not_res, value)
```

Comment:
why not simply put this linear in the method? this complicates the test readability?.

In other hand the test should have a clear returned message if it fails, now it s returning nothing.

Reply:
I kind of agree. I didn't pay too much attention to the tests, because I didn't understand them very well and they passed, so I left them alone.